pkgs/development/tools/build-managers/msbuild/default.nix

```diff
@@ -33,13 +33,19 @@ stdenv.mkDerivation rec {
   ];
 
   # https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=msbuild
-  phases = ["unpackPhase" "buildPhase" "installPhase" "installCheckPhase"];
+  phases = ["unpackPhase" "patchPhase" "buildPhase" "installPhase" "installCheckPhase"];
 
   # https://github.com/NixOS/nixpkgs/issues/38991
   # bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
   LOCALE_ARCHIVE = stdenv.lib.optionalString stdenv.isLinux
       "${glibcLocales}/lib/locale/locale-archive";
 
+  postPatch = ''
+    sed -i -e "/<\/projectImportSearchPaths>/a <property name=\"MSBuildExtensionsPath\" value=\"$out/lib/mono/xbuild\"/>" \
```

Comment:
Can we use substituteInPlace?

Reply:
Well it's inserting above the matched line, rather than replacing it.  It could be done with a substitution, but I'm not sure if it would be clearer.

Reply:
Didn't see that in which case it is fine.